image/image_linux.go

```diff
@@ -719,12 +719,31 @@ func selectAssertionMaxFormats(tsto *tooling.ToolingStore, sysSn *seedwriter.See
 	if err != nil {
 		return err
 	}
-	// XXX take also kernel into account
-	// XXX warning logic
 	maxFormats, _, err := snap.SnapdAssertionMaxFormatsFromSnapFile(snapf)
 	if err != nil {
 		return err
 	}
+	if model.Grade() != asserts.ModelGradeUnset && kSn != nil {
+		// take also kernel into account
+		kf, err := snapfile.Open(kSn.Path)
```

Comment:
It probably does not matter too much as this code doesn't get called a lot, but couldn't we load the kernel max formats once instead or reading it each time?

Reply:
the entire function is called once by the caller, there is flag in the caller code to ensure that

Reply:
Ah right, missed the `assertMaxFormatsSelected`, thank you